quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/PropertyMappers.java

```diff
@@ -116,23 +117,26 @@ public static String canonicalFormat(String name) {
     }
 
     public static String formatValue(String property, String value) {
+        property = removeProfilePrefixIfNeeded(property);
```

Comment:
Finding the right format is easy if you have the property name than its CLI format. I think you can make this change more reliable in regards to the format if you iterate of mappers and transform their names to the CLI format.

Reply:
true. will change :)